### ISSUE TITLE
# FIXME: Actually the first decoder layer should get the correct label from above

### ISSUE THREAD

**Issue by pribadihcr** [tarvaina/tensorflow-ladder#1]
HI @tarvaina,

Did you have fix that?

Thanks

**Comment by tarvaina** [comment]
Hi and thanks for the question,

No, I haven't, but it is not a crucial difference. Ladder should work fine with either the correct label or the predicted label.

However, I should warn you that this repo was a TensorFlow exercise that I never finished and I never got good semisupervised results with it. I don't know what was the cause of that.  One possible culprit is the way we train with minibatches. In this repo (if I remember correctly) we alternate between labeled and unlabeled minibatches. Curious AI's [Theano ladder](https://github.com/CuriousAI/ladder) includes both labeled and unlabeled examples in every minibatch. Also, the hyperparameters might be wrong.

Let me know if you have other questions. If looking for authorative source of Ladder, I urge you to look at the Theano repo I linked above. I don't know if there's a good public Tensorflow implementation of Ladder available.

**Comment by tarvaina** [comment]
Also, have a look at Rinu's implementation. I think it works better. https://github.com/rinuboney/ladder
